pithos/pithos.py

```diff
@@ -889,15 +892,13 @@ def get_album_art(url, tmpdir, *extra):
             song, index = extra
             if tmpdir:
                 try:
-                    # TODO move cache clear call somewhere more sane?
                     self.clear_art_cache()
-                    encoded_filename = (song.artist+song.album).encode('utf-8').hex()[:222]+'.jpeg'
-                    if os.path.exists(os.path.join(self.tempdir, encoded_filename)):
-                        file_url = urllib.parse.urljoin('file://', urllib.parse.quote(encoded_filename))
+                    filename_hash = hashlib.sha256((song.artist+song.album).encode('utf-8')).hexdigest()+'.jpeg'
+                    if os.path.exists(os.path.join(self.tempdir, filename_hash)):
+                        file_url = urllib.parse.urljoin('file://', urllib.parse.quote(filename_hash))
                     else:
```

Comment:
```suggestion
                    file_url = urllib.parse.urljoin('file://', urllib.parse.quote(filename_hash))
                    if not os.path.exists(os.path.join(self.tempdir, filename_hash)):
```

Shouldn't this be like this?

`file_url` is always the same, just write the file if not cached already?

Reply:
Oh actually isn't this path just wrong. It is `os.path.join(self.tempdir, filename_hash)` not just `filename_hash`.

Reply:
Yeah I think I got it this time